File: twosum/twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
	let required = {}

	for (let i = 0; i < nums.length; i++) {
		let diff = target - nums[i]

		if (nums[i] in required) {
			return [required[nums[i]], i]
		}

		required[diff] = i
	}
}

File: twosum/twosum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    differences = Hash.new

    for i in 0...nums.length do
        diff = target - nums[i]
        if differences.key?(nums[i])
            return [differences[nums[i]], i];
        end
        differences[diff] = i
    end
end